diff --git a/dt_totk/lib/agl/include/lyr/aglLayer.h b/dt_totk/lib/agl/include/lyr/aglLayer.h
index 20801e2..d84e3b2 100644
--- a/dt_totk/lib/agl/include/lyr/aglLayer.h
+++ b/dt_totk/lib/agl/include/lyr/aglLayer.h
@@ -1,9 +1,8 @@
 #pragma once
 
 #include <container/seadBuffer.h>
-#include <container/seadListImpl.h>
-#include "container/seadTList.h"
 #include <gfx/seadViewport.h>
+#include <heap/seadDisposer.h>
 #include <mc/seadJob.h>
 #include <prim/seadRuntimeTypeInfo.h>
 #include <prim/seadSafeString.h>
@@ -21,10 +20,10 @@ class Renderer;
 class RenderDisplay;
 class RenderStep;
 
-class Layer {
-    SEAD_RTTI_BASE(Layer)
+class Layer : public sead::IDisposer {
 public:
     virtual ~Layer();
+    SEAD_RTTI_BASE(Layer)
 
     struct CreateArg {
         sead::Heap* heap;
@@ -34,15 +33,19 @@ public:
         u8 _20;
     };
 
+    const char* getName() const { return mLayerName.cstr(); }
+
+#if not LOTUSKIT_HACK_SKIP_PRIVATE
 private:
-    sead::Heap* mContainedHeap;
-    sead::TListNode<sead::Heap*> mHeapListNode;
+#endif
+    //sead::Heap* mContainedHeap;
+    //sead::TListNode<sead::Heap*> mHeapListNode;
     Renderer* mRenderer;
     sead::Viewport mRenderInfoViewport;
     sead::Viewport mRenderViewport;
     sead::DirectCamera* mCamera;
     sead::Projection* mProjection;
-    u16 _88;
+    u16 mRenderFlags;
     u16 _8A;
     char _8C[4];
     u8 _90;
@@ -67,4 +70,4 @@ private:
     char _28[8];
 };
 
-} // namespace agl::lyr
\ No newline at end of file
+} // namespace agl::lyr
