diff --git a/dt_totk/lib/NintendoSDK/include/nvn/nvn_Cpp.h b/dt_totk/lib/NintendoSDK/include/nvn/nvn_Cpp.h
index ec097a4..202499b 100755
--- a/dt_totk/lib/NintendoSDK/include/nvn/nvn_Cpp.h
+++ b/dt_totk/lib/NintendoSDK/include/nvn/nvn_Cpp.h
@@ -1538,7 +1538,7 @@ public:
 
 class VertexAttribState {
 protected:
-    char reserved[8];
+    char reserved[4];
 
 public:
     VertexAttribState& SetDefaults();
