diff --git a/exlaunch/source/lib/util/neon.hpp b/exlaunch/source/lib/util/neon.hpp
index a74d94a..ffb42ad 100644
--- a/exlaunch/source/lib/util/neon.hpp
+++ b/exlaunch/source/lib/util/neon.hpp
@@ -3,6 +3,9 @@
 #include <common.hpp>
 
 namespace exl::util::neon {
+    #ifndef STRINGIFY
+    #define STRINGIFY(x) #x
+    #endif
 
     #define VECTOR_SIZE(size) [[gnu::vector_size(size)]]
     #define VECTOR_NAME(bits, length) F##bits##x##length
@@ -38,4 +41,5 @@ namespace exl::util::neon {
     #undef VECTOR_SIZE
     #undef VECTOR_NAME
     #undef VECTOR_TYPE
-}
\ No newline at end of file
+    #undef STRINGIFY
+}
