diff --git a/subrepo/angelscript/source/as_restore.cpp b/subrepo/angelscript/source/as_restore.cpp
index 26dcaaa..18bc87b 100644
--- a/subrepo/angelscript/source/as_restore.cpp
+++ b/subrepo/angelscript/source/as_restore.cpp
@@ -3227,7 +3227,7 @@ void asCReader::TranslateFunction(asCScriptFunction *func)
 }
 
 asCReader::SListAdjuster::SListAdjuster(asCReader* rd, asDWORD* bc, asCObjectType* listType) :
-	reader(rd), allocMemBC(bc), maxOffset(0), patternType(listType), repeatCount(0), lastOffset(-1), nextOffset(0), nextTypeId(-1), patternNode(0)
+	reader(rd), allocMemBC(bc), maxOffset(0), patternType(listType), repeatCount(0), lastOffset(-1), nextOffset(0), patternNode(0), nextTypeId(-1)
 {
 	asASSERT( patternType && (patternType->flags & asOBJ_LIST_PATTERN) );
 
