diff --git a/subrepo/angelscript/include/scriptstdstring.h b/subrepo/angelscript/include/scriptstdstring.h
index ab2a18e..28028ac 100644
--- a/subrepo/angelscript/include/scriptstdstring.h
+++ b/subrepo/angelscript/include/scriptstdstring.h
@@ -16,7 +16,9 @@
 #include <angelscript.h>
 #endif
 
-#include <string>
+//#include <string>
+#include "exlaunch.hpp"
+using string = lotuskit::String;
 
 //---------------------------
 // Compilation settings
@@ -40,7 +42,7 @@
 
 // This option disables the implicit operators with primitives
 #ifndef AS_NO_IMPL_OPS_WITH_STRING_AND_PRIMITIVE
-#define AS_NO_IMPL_OPS_WITH_STRING_AND_PRIMITIVE 0
+#define AS_NO_IMPL_OPS_WITH_STRING_AND_PRIMITIVE 1
 #endif
 
 BEGIN_AS_NAMESPACE
diff --git a/subrepo/angelscript/source/scriptstdstring.cpp b/subrepo/angelscript/source/scriptstdstring.cpp
index b97c463..508ee75 100644
--- a/subrepo/angelscript/source/scriptstdstring.cpp
+++ b/subrepo/angelscript/source/scriptstdstring.cpp
@@ -10,7 +10,7 @@
 #include <regex>
 
 
-using namespace std;
+//using namespace std;
 
 // This macro is used to avoid warnings about unused variables.
 // Usually where the variables are only used in debug mode.
@@ -21,12 +21,12 @@ using namespace std;
 // cache, so the unordered_map is faster than the ordinary map
 #include <unordered_map>  // std::unordered_map
 BEGIN_AS_NAMESPACE
-typedef unordered_map<string, int> map_t;
+typedef std::unordered_map<string, int> map_t;
 END_AS_NAMESPACE
 #else
 #include <map>      // std::map
 BEGIN_AS_NAMESPACE
-typedef map<string, int> map_t;
+typedef std::map<string, int> map_t;
 END_AS_NAMESPACE
 #endif
 
@@ -680,7 +680,7 @@ static void StringFormat(asIScriptGeneric* gen)
 					break;
 
 #define AS_STRING_FORMAT_IMPL(tid, type) \
-	case tid: result += to_string(*(type*)ref); break
+        case tid: result += std::to_string(*(type*)ref); break
 
 					AS_STRING_FORMAT_IMPL(asTYPEID_INT8, int8_t);
 					AS_STRING_FORMAT_IMPL(asTYPEID_INT16, int16_t);
@@ -714,7 +714,7 @@ static void StringFormat(asIScriptGeneric* gen)
 					else // enums
 					{
 						// TODO: Format enum name
-						result += to_string(*(int*)ref);
+						result += std::to_string(*(int*)ref);
 					}
 				}
 			}
@@ -742,7 +742,7 @@ static void StringScan(asIScriptGeneric* gen)
 {
 	asIScriptEngine* engine = gen->GetEngine();
 
-	stringstream ss(*(string*)gen->GetArgObject(0));
+  std::stringstream ss(*(string*)gen->GetArgObject(0));
 	asUINT scanned = 0;
 
 	for (asUINT i = 1; i < (asUINT)gen->GetArgCount(); ++i)
@@ -971,11 +971,15 @@ static string StringSubString(asUINT start, int count, const string &str)
 // For some reason gcc 4.7 has difficulties resolving the
 // asFUNCTIONPR(operator==, (const string &, const string &)
 // makro, so this wrapper was introduced as work around.
-static bool StringEquals(const std::string& lhs, const std::string& rhs)
+static bool StringEquals(const string& lhs, const string& rhs)
 {
 	return lhs == rhs;
 }
 
+static string StringConcat(const string& lhs, const string& rhs) {
+    return lhs + rhs;
+}
+
 void RegisterStdString_Native(asIScriptEngine *engine)
 {
 	int r = 0;
@@ -1003,7 +1007,7 @@ void RegisterStdString_Native(asIScriptEngine *engine)
 	// Need to use a wrapper for operator== otherwise gcc 4.7 fails to compile
 	r = engine->RegisterObjectMethod("string", "bool opEquals(const string &in) const", asFUNCTIONPR(StringEquals, (const string &, const string &), bool), asCALL_CDECL_OBJFIRST); assert( r >= 0 );
 	r = engine->RegisterObjectMethod("string", "int opCmp(const string &in) const", asFUNCTION(StringCmp), asCALL_CDECL_OBJFIRST); assert( r >= 0 );
-	r = engine->RegisterObjectMethod("string", "string opAdd(const string &in) const", asFUNCTIONPR(operator +, (const string &, const string &), string), asCALL_CDECL_OBJFIRST); assert( r >= 0 );
+	r = engine->RegisterObjectMethod("string", "string opAdd(const string &in) const", asFUNCTIONPR(StringConcat, (const string &, const string &), string), asCALL_CDECL_OBJFIRST); assert( r >= 0 );
 
 	// The string length can be accessed through methods or through virtual property
 	// TODO: Register as size() for consistency with other types
diff --git a/subrepo/angelscript/source/scriptstdstring_utils.cpp b/subrepo/angelscript/source/scriptstdstring_utils.cpp
index afb7c55..bddf57f 100644
--- a/subrepo/angelscript/source/scriptstdstring_utils.cpp
+++ b/subrepo/angelscript/source/scriptstdstring_utils.cpp
@@ -4,7 +4,7 @@
 #include <stdio.h>
 #include <string.h>
 
-using namespace std;
+//using namespace std;
 
 BEGIN_AS_NAMESPACE
 
