diff --git a/angelscript/source/as_compiler.cpp b/angelscript/source/as_compiler.cpp
index ab86ab3..ba73075 100644
--- a/angelscript/source/as_compiler.cpp
+++ b/angelscript/source/as_compiler.cpp
@@ -5209,7 +5209,7 @@ void asCCompiler::CompileForEachStatement(asCScriptNode* node, asCByteCode* bc)
 			asIScriptFunction* f = engine->scriptFunctions[funcs[i]];
 			asDWORD flags;
 			int paramTid;
-			if (f->GetParamCount() == 1 && f->GetReturnTypeId(&flags) == asTYPEID_BOOL && !(flags && asTM_INOUTREF) && f->GetParam(0, &paramTid) >= 0 && paramTid == iterTid)
+			if (f->GetParamCount() == 1 && f->GetReturnTypeId(&flags) == asTYPEID_BOOL && !(flags && asTM_INOUTREF > 0) && f->GetParam(0, &paramTid) >= 0 && paramTid == iterTid)
 			{
 				if (opForEndId != 0 && !isConstRange)
 				{
